.github/workflows/pdm-lock-automation.yaml

```diff
@@ -5,127 +5,68 @@ on:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          if [ "${{ matrix.directory }}" != "root" ]; then
-            cd ${{ matrix.directory }}
-          fi
+          chmod +x ./scripts/pdm-lock.sh
 
-          if [ ! -d ".venv" ]; then
-            echo 'Creating virtual environment inside "${{ matrix.directory }}".'
-            pdm venv create -w virtualenv --with-pip
-          else
-            echo "Virtual environment already exists."
-          fi
-
-          source .venv/bin/activate
-
-          echo "Generating/Updating pdm.lock..."
-          pdm lock -G :all -v
-
-          echo "Checking pdm lock..."
-          pdm lock --check
-
-          echo "Stashing changes if there are any..."
-          git stash || echo "No changes to stash, continuing..."
+          # Get the input from the workflow or use the default value
+          dirs="${{ github.event.inputs.directories }}"
 
-          echo "Pulling remote ${{ github.head_ref }} branch..."
-          git pull origin ${{ github.head_ref }} --allow-unrelated-histories --no-rebase
+          # Check if directories input is empty
+          if [[ -z "$dirs" ]]; then
+            # No directories input given, run the script without arguments (process all directories)
+            echo "No directories specified, running on all dirs listed in scripts/pdm-lock.sh"
 
-          echo "Trying to pop stash if any..."
-          if git stash list | grep -q 'stash@{0}'; then
-            echo "Popping stash..."
-            git stash pop || echo "Failed to pop stash. Continue with other changes."
+            ./scripts/pdm-lock.sh
           else
-            echo "No stash found, skipping pop."
-          fi
-
-          if git diff --quiet; then
-            echo "No changes between the current branch and the ${{ github.head_ref }} branch."
-            echo "Nothing to commit"
-          else
-            echo "Changes detected between the current branch and the ${{ github.head_ref }} branch."
+            # Convert comma-separated list into an array of directories
+            IFS=',' read -r -a dir_array <<< "$dirs"
 
-            echo "Adding pdm lock changes..."
-            git add pdm.lock
+            # Print directories being processed
+            echo "Processing specified directories: ${dir_array[@]}"
 
-            echo "Committing and pushing pdm lock ..."
-            git commit -m "Update pdm.lock for ${{ matrix.directory }}"
-            git push origin HEAD:${{ github.head_ref }}
+            # Pass directories as command-line arguments to the script
+            ./scripts/pdm-lock.sh "${dir_array[@]}"
           fi
+        shell: bash
+
+      - name: Commit pdm.lock changes
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: Commit pdm.lock changes
+          commit_user_name: pdm-lock-automation[bot]
+          commit_user_email: pdm-lock-automation-bot@unstract.com
```

Comment:
The problem with the earlier workflow was it was generating stale pdm lock.
The reason is when multiple jobs run, the commit was not handling the pdm lock conflict, resulting in stale pdm.
Will this resolve this?

Reply:
@Kirtiman I was under the impression that the issue was with how we commit the changes due to conflicts from the stash and upstream. If that's the case - this will ensure that we commit only once after all the changes. Unsure whether it'll cause issues when multiple workflows run from different branches - based on the frequency of such issues we can tackle it I believe

Are you talking about python dependency conflicts here? If that's the case this action itself might not be helpful and we'll have to intervene. Let me try to check if we still commit in case of such conflicts

Reply:
@chandrasekharan-zipstack since we are committing only once, it should solve the issue. Resolving and approving.